packages/fx-core/src/component/generator/apiSpec/helper.ts

```diff
@@ -966,6 +1008,24 @@ async function updateCodeForCustomApi(
       .replace("{{OPENAPI_SPEC_PATH}}", openapiSpecFileName)
       .replace("// Replace with action code", actionsCode.join("\n"));
     await fs.writeFile(indexFilePath, updateIndexFileContent);
+  } else if (language === ProgrammingLanguage.PY) {
+    // Update code in bot.py
+    const actionsCode = [];
+    const codeTemplate = ActionCode["python"];
+    for (const item of specItems) {
+      const code = codeTemplate
+        .replace(/{{operationId}}/g, item.item.operationId!)
+        .replace(/{{pathUrl}}/g, item.pathUrl)
+        .replace(/{{method}}/g, item.method);
+      actionsCode.push(code);
+    }
+
+    const botFilePath = path.join(destinationPath, "src", "bot.py");
+    const botFileContent = (await fs.readFile(botFilePath)).toString();
+    const updateBotFileContent = botFileContent
+      .replace("{{OPENAPI_SPEC_PATH}}", openapiSpecFileName)
+      .replace("# Replace with action code", actionsCode.join("\n"));
```

Comment:
How about using similar placeholder {{ACTION_CODE}}

Reply:
This part follow same pattern as ts/js. We may update them together in another pr if needed.